app/models/risk_condition.rb

```diff
@@ -11,44 +11,107 @@ class RiskCondition < ApplicationRecord
   # :no_submissions => :no_submissions_threshold
   # :low_grades => :grade_threshold, :count_threshold
 
-  def self.create_condition_for_course_with_type(course_id, type, params)
+  def self.create_condition_for_course_with_type(course_id, type, params, version)
   	# parameter check shouldn't surface to user and is for sanity check during development only
-    if (type == :grace_day_usage && (params[:grace_day_threshold].nil? || params[:date].nil? || params.length != 2)) ||
-       (type == :grade_drop && (params[:percentage_drop].nil? || params[:consecutive_counts].nil? || params.length != 2)) ||
-       (type == :no_submissions && (params[:no_submissions_threshold].nil? || params.length != 1)) ||
-       (type == :low_grades && (params[:grade_threshold].nil? || params[:count_threshold].nil? || params.length != 2))
+    if (type == 1 && (params[:grace_day_threshold].nil? || params[:date].nil? || params.length != 2)) ||
```

Comment:
this seems like a regression? Was there an issue when it comes to using an enum?

Reply:
I believe comparing an int against an enum hash doesn't work as intended. But enum does help with drawing correspondence between the hash and the int.

Reply:
I think that is likely due to the fact that you have added :no_condition_selected to the front of the enum. The number of the enum is ordered based on the index in the array.

A solution to your issue would be to specific the enum index specifically in this manner
`  enum status: { active: 0, archived: 1 }`

Reply:
I tried the hash notation instead of array and it still doesn't work. Actually, if you take a look at the previous code, 1 represents grace_day_usage and so on so forth so 0 reserved for no condition selected has already been accounted for.

Reply:
hmm okay. Anyway this doesn't really affect the functionality, so we can just leave this at that. Can come back to address this in a future PR.